### PR TITLE
Add IGDB scraper

### PR DESCRIPTION
**Pull request by Gorlah**
This PR adds a new `goty-scraper` module for scraping game titles and alternative titles from IGDB and writing them to a text file called `game_titles.json`.

The main `goty-server` module reads the text file and exposes a simple endpoint for searching on all and alternative titles.

### CODE REVIEW

File: goty-scraper/src/main/java/com/aleinin/gotyscraper/IGDBScraper.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
+package com.aleinin.gotyscraper;
+
+import com.api.igdb.apicalypse.APICalypse;
+import com.api.igdb.exceptions.RequestException;
+import com.api.igdb.request.IGDBWrapper;
+import com.api.igdb.utils.Endpoints;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.Builder;
```
Comment (aleinin) [inline goty-scraper/src/main/java/com/aleinin/gotyscraper/IGDBScraper.java:10]: image using java with lombok, and not just Kotlin. 

( <3 )

File: goty-server/src/main/java/com/aleinin/goty/TitleService.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        }
+    }
+
+    @Value
+    @Builder
+    private static class Game {
+
+        int id;
+        String name;
+        List<String> names;
```
Comment (aleinin) [inline goty-server/src/main/java/com/aleinin/goty/TitleService.java:55]: in what context would a game ave a name and a list of names?

Comment (Gorlah) [reply goty-server/src/main/java/com/aleinin/goty/TitleService.java:55]: The IGDB API additionally returns "alternative titles" for a game. So for instance "TES5" and "TESV" are alternative titles for Skyrim. What I'm doing is including both the "primary" and "alternative" titles in the "names" field and searching on it. When there's a match, return the primary title only. So if a search was done for "TES5" the response would return "The Elder Scrolls V: Skyrim". 

I am actually working on some changes right now to make this all a little more clear and to improve the searching functionality.

Comment (aleinin) [reply goty-server/src/main/java/com/aleinin/goty/TitleService.java:55]: Yeah I'd recommend calling it "aliases" or "alternative names" or something. Your logic makes sense I just couldn't infer that from what was there

Comment (Gorlah) [reply goty-server/src/main/java/com/aleinin/goty/TitleService.java:55]: "aliases" is good. 👍 

Names are hard lol

Comment (Gorlah) [reply goty-server/src/main/java/com/aleinin/goty/TitleService.java:55]: @aleinin I will fix it in my next pr. I'm already pretty far along there.

**Review by aleinin — APPROVED** [review]

**Pull request merged**